src/ext.rs

```diff
@@ -832,6 +858,31 @@ fn timeout2ms(dur: DWORD) -> Option<u32> {
 }
 
 #[cfg(any(feature = "nightly", feature = "duration"))]
+fn linger2dur(linger_opt: linger) -> Option<Duration> {
+    let dur = Duration::from_secs(linger_opt.l_linger as u64);
```

Comment:
Perhaps this should be:

```rust
if linger.l_onoff != 0 {
    Some(Duration::new(linger.l_linger as u64))
} else {
    None
}
```

That is, whether or not this is `None` is based on `l_onoff`, right?